File: src/core/java/net/remmintan/mods/minefortress/core/interfaces/tasks/IServerTaskManager.java
package net.remmintan.mods.minefortress.core.interfaces.tasks;

import net.minecraft.server.network.ServerPlayerEntity;
import net.remmintan.mods.minefortress.core.interfaces.server.IServerManager;

import java.util.List;
import java.util.UUID;

public interface IServerTaskManager extends IServerManager {
    void addTask(ITask task, List<Integer> selectedPawnIds, ServerPlayerEntity player);


    void cancelTask(UUID id, ServerPlayerEntity player);
}

File: src/core/java/net/remmintan/mods/minefortress/core/interfaces/tasks/IServerTaskManager.kt
package net.remmintan.mods.minefortress.core.interfaces.tasks

import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.util.math.BlockPos
import net.remmintan.mods.minefortress.core.interfaces.server.IServerManager

interface IServerTaskManager : IServerManager {
    fun addTask(task: IBaseTask, selectedPawnIds: List<Int>, player: ServerPlayerEntity)
    fun cancelTask(pos: BlockPos, player: ServerPlayerEntity)
}
